float32 doa
float32 elevation

